### DISCUSSION TITLE
Does window.dispatchReactUnityEvent() return a value?

### DISCUSSION THREAD

**Discussion by kevin-guertin** [jeffreylanters/react-unity-webgl#539] [Question and Answer] [answered]
Hi,

I have yet to upgrade from v8 to v9 for my project at work and I'm currently evaluating the impact it would have on our codebases before we try the upgrade. 

One thing that doesn't seem clear is whether the `window.dispatchReactUnityEvent` can return a value if the React code returns a value. I've looked through the docs and discussions here and I haven't been able to find this functionality in any **v9** code.

For example, here is some redacted **v8** code:

```javascript
mergeInto(LibraryManager.library, { 

  GetElectricalCodeJS: function() {
    var returnStr = ReactUnityWebGL.GetElectricalCode();
    var bufferSize = lengthBytesUTF8(returnStr) + 1;
    var buffer = _malloc(bufferSize);
    stringToUTF8(returnStr, buffer, bufferSize);
    return buffer;
  },
});
```

This code allows Unity to get a value from React. We have multiple functions like this and it's not clear if the following **v9** code will work:

```javascript
mergeInto(LibraryManager.library, { 

  GetElectricalCodeJS: function() {
    var returnStr = window.dispatchReactUnityEvent("GetElectricalCode");
    var bufferSize = lengthBytesUTF8(returnStr) + 1;
    var buffer = _malloc(bufferSize);
    stringToUTF8(returnStr, buffer, bufferSize);
    return buffer;
  },
});
```
To be clear, the `GetElectricalCode()` in React returns a value and this works fine in v8.

If anyone can answer, it would be greatly appreciated. Thanks!

**Comment by kevin-guertin** [comment, marked as answer]
I think I answered my own question by looking at the actual code for `dispatchReactUnityEvent`.  Doh!

```typescript
const dispatchReactUnityEvent = (
  eventName: string,
  ...parameters: ReactUnityEventParameter[]
): ReactUnityEventParameter => {
  // Loops through all of the mounted event systems and dispatches the event.
  // In case there are multiple event systems, the return value origin is
  // undefined.
  let returnValue: ReactUnityEventParameter = undefined;
  mountedEventDispatchers.forEach((dispatchEvent) => {
    returnValue = dispatchEvent(eventName, ...parameters);
  });
  return returnValue;
};
```
Should have an example for this in the docs.

**Reply by jeffreylanters** [reply]
I will take a look at adding it to the documentation
